From 5492f773f00dceba009f6dc65b3b8f50b14ebe64 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Thu, 30 Sep 2021 13:53:06 +0200
Subject: [PATCH 0221/1695] core: dt: export add_probe_node_by_compat function

Export the function that can be used to add nodes to the
device list that will be probed during boot.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4cb560ead64377a977a7df85a28020b2945f7cdf core: dt: export add_probe_node_by_compat function]
Change-Id: I58b1354632e122e1a43a86963574eedcce043a7c
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/include/kernel/dt_driver.h | 8 ++++++++
 core/kernel/dt_driver.c         | 4 ++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/core/include/kernel/dt_driver.h b/core/include/kernel/dt_driver.h
index 0eef55db0..e440e6f5f 100644
--- a/core/include/kernel/dt_driver.h
+++ b/core/include/kernel/dt_driver.h
@@ -265,6 +265,14 @@ int fdt_get_dt_driver_cells(const void *fdt, int nodeoffset,
  */
 TEE_Result dt_driver_maybe_add_probe_node(const void *fdt, int nodeoffset);
 
+/* Add a matching node to the probe list
+ * @compat: compat to search for matching the node
+ *
+ * Return appropriate TEE error code
+ */
+TEE_Result add_probe_node_by_compat(const void *fdt, int node,
+				    const char *compat);
+
 #ifdef CFG_DT_DRIVER_EMBEDDED_TEST
 /*
  * Return TEE_ERROR_NOT_IMPLEMENTED if test are not implemented
diff --git a/core/kernel/dt_driver.c b/core/kernel/dt_driver.c
index fe3169633..290b18ea3 100644
--- a/core/kernel/dt_driver.c
+++ b/core/kernel/dt_driver.c
@@ -709,8 +709,8 @@ static TEE_Result add_node_to_probe(const void *fdt, int node,
  *	  TEE_ERROR_ITEM_NOT_FOUND if no matching driver
  *	  TEE_ERROR_OUT_OF_MEMORY if heap is exhausted
  */
-static TEE_Result add_probe_node_by_compat(const void *fdt, int node,
-					   const char *compat)
+TEE_Result add_probe_node_by_compat(const void *fdt, int node,
+				    const char *compat)
 {
 	TEE_Result res = TEE_ERROR_ITEM_NOT_FOUND;
 	const struct dt_driver *dt_drv = NULL;
-- 
2.47.2

